package/kernel/linux/modules/netsupport.mk

```diff
@@ -1250,3 +1250,33 @@ define KernelPackage/netlink-diag/description
 endef
 
 $(eval $(call KernelPackage,netlink-diag))
+
+
+define KernelPackage/wireguard
+  SECTION:=kernel
+  CATEGORY:=Kernel modules
+  SUBMENU:=Network Support
+  TITLE:=WireGuard kernel module
+  DEPENDS:= \
+	  +kmod-crypto-blake2s \
+	  +kmod-crypto-chacha20poly1305 \
+	  +kmod-crypto-curve25519 \
```

Comment:
These three modules are needed on the disk image, but the kconfig symbol is forced by the wireguard module, so the kconfig stuff above might not be necessary?

Reply:
Yeah I think we want those here because of how openwrt builds things (maybe for .ipk installable modules?), but I'm not 100% sure either, just following examples of other kernel modules.

Reply:
But this still forces the nonexistent kmod-crypto-blake2s on an ARM build.

Reply:
Why nonexistent? WireGuard still needs the libblake2s code.

Reply:
@anomeome I took out the arch-specific blake2s-arm that is not in 5.10 - does it still fail for you with latest rev of this branch?

Reply:
Actually a new one now

`ERROR: module '/home/kc/wrtpac/source/build_dir/target-arm_cortex-a9+vfpv3_musl_eabi/linux-mvebu_cortexa9/linux-5.10.17/arch/arm/crypto/nhpoly1305-neon.ko' is missing`

Reply:
Okay, please try with latest code - I fixed the error above, but don't have a way to test build against your target. Open a new discussion if still broken.